cs-property-service/src/main/java/org/egov/cpt/service/DemandGenerationService.java

```diff
@@ -0,0 +1,131 @@
+package org.egov.cpt.service;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.egov.cpt.config.PropertyConfiguration;
+import org.egov.cpt.models.DemandCriteria;
+import org.egov.cpt.models.Property;
+import org.egov.cpt.models.PropertyCriteria;
+import org.egov.cpt.models.RentDemand;
+import org.egov.cpt.producer.Producer;
+import org.egov.cpt.repository.PropertyRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Service
+public class DemandGenerationService {
+
+	private PropertyRepository propertyRepository;
+	
+	private Producer producer;
+	
+	private PropertyConfiguration config;
+	
+	private RentCollectionService rentCollectionService;
+
+	private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("d/MM/yyyy");
+	@Autowired
+	public DemandGenerationService(PropertyRepository propertyRepository, Producer producer, PropertyConfiguration config) {
+		this.propertyRepository = propertyRepository;
+		this.producer = producer;
+		this.config = config;
+	}
+
+	public void createDemand(DemandCriteria demandCriteria) {
+		
+		PropertyCriteria propertyCriteria = new PropertyCriteria();
+		List<Property> propertyList = propertyRepository.getProperties(propertyCriteria);
+		// System.out.println(propertyList);
+
+		propertyList.forEach(property -> {
+			propertyCriteria.setPropertyId(property.getId());
```

Comment:
Make sure one crash (exception) for one property's calculation does not stop the process for all properties. Catch exception, log it and move onto the next property.

Reply:
try catch added